### PR TITLE
Speaker and event details to firestore

### PR DESCRIPTION
**Pull request by quiro91**
## Problem

We are moving more data to Firestore

## Solution

Speakers and event details now come from Firestore

### Test(s) added 

None unfortunately

### Paired with 

Nobody

### CODE REVIEW

File: app/src/main/java/net/squanchy/schedule/ScheduleService.kt
Comment (fourlastor) [inline app/src/main/java/net/squanchy/schedule/ScheduleService.kt:44]: can you inline this now? or does it lose the types?

File: app/src/main/java/net/squanchy/service/firestore/FirestoreMappers.kt
Comment (fourlastor) [inline app/src/main/java/net/squanchy/service/firestore/FirestoreMappers.kt]: shouldn't this be in line with others in the form of `experienceLevel?.toExperienceLevel().optional()`?

File: app/src/main/java/net/squanchy/service/repository/firebase/FirebaseEventRepository.kt
Comment (fourlastor) [inline app/src/main/java/net/squanchy/service/repository/firebase/FirebaseEventRepository.kt]: can we make this a function and refer to it with `::` instead?

File: app/src/main/java/net/squanchy/service/repository/firebase/FirebaseEventRepository.kt
Comment (fourlastor) [inline app/src/main/java/net/squanchy/service/repository/firebase/FirebaseEventRepository.kt]: same here

File: app/src/main/java/net/squanchy/service/repository/firebase/FirestoreSpeakerRepository.kt
Comment (fourlastor) [inline app/src/main/java/net/squanchy/service/repository/firebase/FirestoreSpeakerRepository.kt]: this has to go with a `dbService.speaker(speakerId)`, we have firestore so we can do querying :)

**Review by fourlastor — CHANGES_REQUESTED** [review]
A few comments, the most important being using querying for the speakers, the rest is mostly nitpicking

File: app/src/main/java/net/squanchy/service/firestore/FirestoreMappers.kt
Comment (rock3r) [inline app/src/main/java/net/squanchy/service/firestore/FirestoreMappers.kt]: These should be very easy to test! Can we have tests plz?

File: app/src/main/java/net/squanchy/service/repository/firebase/FirebaseEventRepository.kt
Comment (rock3r) [inline app/src/main/java/net/squanchy/service/repository/firebase/FirebaseEventRepository.kt:18]: :x: using the favourites from RTDB is most likely _not_ going to be working. We should leave them empty for now! There's a TODO for them in #367.

**Review by rock3r — CHANGES_REQUESTED** [review]
Great one, except that we don't have favourites for now and we should NOT use the ones from RTDB as they're an entirely different thing

File: app/src/main/java/net/squanchy/schedule/ScheduleService.kt
Comment (quiro91) [reply app/src/main/java/net/squanchy/schedule/ScheduleService.kt:44]: Yeah it does lose the type

File: app/src/main/java/net/squanchy/service/firestore/FirestoreMappers.kt
Comment (quiro91) [reply app/src/main/java/net/squanchy/service/firestore/FirestoreMappers.kt]: Makes sense 👍 
Edit: the issue here is that experienceLevel is a String and I don't really want to expose an extension function over the String type. Shall we have it private?

File: app/src/main/java/net/squanchy/service/firestore/FirestoreMappers.kt
Comment (quiro91) [reply app/src/main/java/net/squanchy/service/firestore/FirestoreMappers.kt]: Will do a PR on top of this one for testing, working on it

File: app/src/main/java/net/squanchy/service/firestore/FirestoreMappers.kt
Comment (fourlastor) [reply app/src/main/java/net/squanchy/service/firestore/FirestoreMappers.kt]: Yeah private is cool to me

**Comment by quiro91** [comment]
@rock3r @fourlastor ready for another review. Also, before merging take a look at #453 which adds test for a lot of the Firestore model

File: app/src/main/java/net/squanchy/service/repository/firebase/FirebaseEventRepository.kt
Comment (quiro91) [reply app/src/main/java/net/squanchy/service/repository/firebase/FirebaseEventRepository.kt:18]: Fixed

**Comment by rock3r** [comment]
Do not merge until #453 is shipped

**Review by rock3r — APPROVED** [review]
Nice one, just waiting for the tests to be merged and then we can ship for what I am concerned

**Review by fourlastor — APPROVED** [review]

**Comment by rock3r** [comment]
Tests have been merged, now we can 🚢 once green

**Pull request merged**
